### DISCUSSION TITLE
Type error: Module '"@prisma/client"' has no exported member

### DISCUSSION THREAD

**Discussion by morganwestcoding** [prisma/prisma#22980] [Q&A]
### Bug description

trying to run docker build . with [this repository ](https://github.com/morganwestcoding/forme-nextjs)

When i run it It goes to step => ERROR [6/6] RUN npm run build 

Then I get this error:

```
 ./src/app/types/index.ts:1:9
30.53 Type error: Module '"@prisma/client"' has no exported member 'Listing'.
30.53 
30.53 > 1 | import {Listing , User, Reservation, Post, Service} from '@prisma/client'
30.53     |         ^
30.53   2 |
30.53   3 | export type SafeService = {
30.53   4 |   id: string;
```

I've checked my node modules and the models are in `node_modules/.prisma/client/index.d.ts`

in my `node_modules/@prisma/client/index.d.ts` I have `export * from '.prisma/client/default'`

Can someone please help me i've been trying to resolve this for days, and extra points if you help me succsessfully launc

docker build . and help me deploy to vercel

I've removed my node modules and package-lock.json and npm installed

I've tried npx prisma generate 

Nothing has worked and i am still getting this error

<img width="678" alt="Screenshot 2024-02-03 at 2 30 44 PM" src="https://github.com/prisma/prisma/assets/17777272/55934769-b223-4d45-add7-47b0de3d3715">

### Expected behavior

I expected the docker build . to successfully create a container image so i can deploy to vercel.

### Prisma information

```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

model User {
  id             String    @id @default(auto()) @map("_id") @db.ObjectId
  name           String?
  email          String?   @unique
  emailVerified  DateTime?
  image          String?
  hashedPassword String?
  createdAt      DateTime  @default(now())
  updatedAt      DateTime  @updatedAt
  favoriteIds    String[]  @db.ObjectId
  accounts     Account[]
  listings     Listing[]
  reservations Reservation[]
  posts          Post[]         // This line represents the relation to the Post model
}

model Account {
  id                String  @id @default(auto()) @map("_id") @db.ObjectId
  userId            String  @db.ObjectId
  type              String
  provider          String
  providerAccountId String
  refresh_token     String? @db.String
  access_token      String? @db.String
  expires_at        Int?
  token_type        String?
  scope             String?
  id_token          String? @db.String
  session_state     String?

  user User @relation(fields: [userId], references: [id], onDelete: Cascade)

  @@unique([provider, providerAccountId])
}

model Listing {
  id            String   @id @default(auto()) @map("_id") @db.ObjectId
  title         String
  description   String
  imageSrc      String
  createdAt     DateTime @default(now())
  category      String
  locationValue String
  userId        String   @db.ObjectId
  services    Service[]
  user         User          @relation(fields: [userId], references: [id], onDelete: Cascade)
  reservations Reservation[]
  
  
}

model Reservation {
  id         String   @id @default(auto()) @map("_id") @db.ObjectId
  userId     String   @db.ObjectId
  listingId  String   @db.ObjectId
  startDate  DateTime
  endDate    DateTime
  totalPrice Int
  createdAt  DateTime @default(now())

  user    User    @relation(fields: [userId], references: [id], onDelete: Cascade)
  listing Listing @relation(fields: [listingId], references: [id], onDelete: Cascade)
}

model Service {
  id          String   @id @default(auto()) @map("_id") @db.ObjectId
  serviceName String
  price       Int
  category    String
  listingId   String   @db.ObjectId
  listing     Listing  @relation(fields: [listingId], references: [id], onDelete: Cascade)
}

model Post {
  id         String   @id @default(auto()) @map("_id") @db.ObjectId
  content    String
  imageSrc   String?
  location   String?
  tag        String?
  photo      String?
  category   String
  userId     String   @db.ObjectId
  createdAt  DateTime @default(now())

  user       User     @relation(fields: [userId], references: [id])

  // Include other fields and relations, but ensure 'userId' is not duplicated
}


model Category {
  id            String   @id @default(auto()) @map("_id") @db.ObjectId
  name          String
  description   String?
}

```

### Environment & setup

MacOS
MongoDb
Node 18.x

### Prisma Version

```
"prisma": "^5.9.1",
"@prisma/client": "^5.9.1",
```

**Comment by morganwestcoding** [comment]
I resolved this but can you please help me out with this issue
https://github.com/vercel/next.js/issues/61697#issue-2119938488

**Reply by ludralph** [reply]
Hi @morganwestcoding 👋 

I'm glad to hear that you were able to resolve the original issue. With regards to the newer issue, I do think it's a next js specific issue and not Prisma related.

**Reply by morganwestcoding** [reply]
Ah good to know ludralph i opened a new issue under the vercel thread [here](https://github.com/vercel/next.js/issues/61697#issue-2119938488) if youre able to help

**Reply by Rakhimi** [reply]
Hi @morganwestcoding,
How did you resolve this issue? I'm facing a similar error

**Reply by Harmaton** [reply]
Hi @morganwestcoding, how did you resolve the first issue?

**Reply by derharry** [reply]
Hi @morganwestcoding , I'm facing similar issue. Can you please share how you solved it to complete this post? thanks.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
